README.rst

```diff
@@ -1317,6 +1317,27 @@ description of nested values.
    end
 
 
+OneOfValidator
+-------------
+
+// TODO: More info here
+
+.. code:: ruby
+
+   param :pet, :one_of, :desc => "Pet info" do
+     # Param names do not have an effect on validation functionality, but they are included in
+     # OpenAPI schema output
+     param :dog, Hash do
+       param :type, ["dog"], required: true
```

Comment:
If I follow your PR description correctly, does this mean the OpenAPI generator will be able to give us:
```ts
enum PetTypes {
  DOG = 'dog',
  CAT = 'cat'
}

interface PstPetParams {
  type: PetTypes
}
```
from the result of this work?

Reply:
No, not quite. I'm working on improving the documentation here to explain how discriminators should be used. `openapi-generator` won't produce an `enum` for the possible discriminator values, it will just use string literals. See [here](https://github.com/instacart/carrot/blob/cb10edf961c7980f5f0ab46ca2c65e796a0f645e/ads/ads-web/src/client/service/openapi/__codegen__/models/PutDisplayBrandPagesIdParamBlocks.ts) for what the generated code looks like.

Reply:
Got it. FWIW this is definitely powerful enough:
https://github.com/instacart/carrot/blob/cb10edf961c7980f5f0ab46ca2c65e796a0f645e/ads/ads-web/src/client/service/openapi/__codegen__/models/PutDisplayBrandPagesIdParamBlocks.ts#L30-L32
to give us really strong type security, so I don't think getting some kind of inferred enum really matters. That's just dev QOL after that.

Thanks for clarifying!